src/d3a/models/market/two_sided_pay_as_bid.py

```diff
@@ -78,7 +78,8 @@ def accept_bid(self, bid: Bid, energy: float = None,
         energy = market_bid.energy if energy is None else energy
         if trade_rate is None:
             trade_rate = market_bid.price / market_bid.energy
-        assert trade_rate <= (market_bid.price / market_bid.energy)
+        assert trade_rate <= (market_bid.price / market_bid.energy) + 0.0001, \
```

Comment:
No need for this to be fixed in this PR (I will probably incorporate it on the 2-sided market story), but floating-point threshold values are scattered throughout our code (a quick search for 0.000 will immediately reveal quite a lot of these occurrences). I will take the time to refactor these to a common threshold variable for the whole project.

Reply:
Integration tests are failing after removing floating threshold. So, should I use rounding, keep it the way it is or any alternative suggestion?

Reply:
Leave it as it is. I am not suggesting to remove it, I will add a corresponding refactoring in the two-sided market PR.